lib/nostrum/struct/channel.ex

```diff
@@ -1,27 +1,125 @@
 defmodule Nostrum.Struct.Channel do
-  @moduledoc ~S"""
+  @moduledoc """
+  Struct representing a channel.
 
-  A `Nostrum.Struct.Channel` represents a Guild or DM channel.
+  There are currently eleven channel types within discord, ranging from direct messages between two users to server announcement channels pushing a message to thousands. This module also contains several helper functions for handling channels within your applcation.
 
-  More information can be found on the
-  [Discord API Channel Documentation](https://discord.com/developers/docs/resources/channel#channel-object).
+  ## Channel Struct
 
-  ## Mentioning Channels in Messages
+  The channel struct is used by Nostrum to represent a _Discord Channel Object_. More information can be found on the [Discord API Channel Documentation](https://discord.com/developers/docs/resources/channel#channels-resource).
 
-  A `Nostrum.Struct.Channel` can be mentioned in message content using the `String.Chars`
-  protocol or `mention/1`.
+  The struct can have one of several forms depending on the type of channel. You can read more about the individual channel types [below](#module-channel-types).
+
+  A typical channel would appear as:
 
   ```elixir
-  channel = %Nostrum.Struct.Channel{id: 381889573426429952}
-  Nostrum.Api.create_message!(184046599834435585, "#{channel}")
-  %Nostrum.Struct.Message{content: "<#381889573426429952>"}
+  channel = %Nostrum.Struct.Channel{
+      guild_id: 766435015768539156,
+      id: 827333533688397865,
+      name: "announcements",
+      nsfw: false,
+      permission_overwrites: [],
+      position: 1,
+      type: 5,
+      }
+  ```
+
+  The channel struct implements `String.Chars` protocol through the `mention/1` function. This example uses our channel from the previous code block.
+
+  ```elixir
+  channel |> to_string()
+  "<#766435015768539156>"
+  ```
+
+  ## Channel Cache
 
-  channel = %Nostrum.Struct.Channel{id: 280085880452939778}
-  Nostrum.Api.create_message!(280085880452939778, "#{Nostrum.Struct.Channel.mention(channel)}")
-  %Nostrum.Struct.Message{content: "<#280085880452939778>"}
+  The [`ChannelCache`](`Nostrum.Cache.ChannelCache`) module provides functionality for you to retreive information about any channel that your application can see. It provides two functions: [`get()/1`](`Nostrum.Cache.ChannelCache.get()/1`) and [`get!()/1`](`Nostrum.Cache.ChannelCache.get!()/1`).
```

Comment:
These functions need to be documented & linked without `()` to make `mix docs` link to them properly:
```suggestion
  The [`ChannelCache`](`Nostrum.Cache.ChannelCache`) module provides functionality for you to retreive information about any channel that your application can see. It provides two functions: [`get/1`](`Nostrum.Cache.ChannelCache.get/1`) and [`get!/1`](`Nostrum.Cache.ChannelCache.get!/1`).
```